lib/iaas/BackupStore.js

```diff
@@ -210,8 +220,12 @@ class BackupStore {
           throw new UnprocessableEntity(`Backup '${backup_guid}' is still in process`);
         }
         if (!options.force && data.trigger === CONST.BACKUP.TRIGGER.SCHEDULED) {
-          const retentionStartDate = this.filename.isoDate(moment()
+          let retentionStartDate = this.filename.isoDate(moment()
             .subtract(config.backup.retention_period_in_days, 'days').toISOString());
+          if (options.only_snapshot === 'true') {
+            retentionStartDate = this.filename.isoDate(moment()
```

Comment:
Yes. we can have another log which will be logged when the `only_snapshot` is `true` and a snapshot is being deleted. It can`Deletion of snapshot not permitted within retention period of ${config.backup.azure_cost_reduction_retention_period}`